contextIO.go

```diff
@@ -149,5 +171,13 @@ func (c *ContextIO) DoJSON(method, q string, params url.Values, body *string) (j
 	}
 	defer response.Body.Close()
 	json, err = ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.StatusCode != http.StatusOK {
```

Comment:
This I think is fair criticism, but on the other hand it is sad to use the DoJSON function and get `incorrect parsing` thrown when what is returned from the DoJSON is something like `{"type":"error","value":"limit cannot exceed 100"}`. 

Is there some sort of middle ground, or is it always up to the user to try to parse out the error?